server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -1826,6 +1827,24 @@ public static IndexMetadata legacyFromXContent(XContentParser parser) throws IOE
                     } else if ("mappings".equals(currentFieldName)) {
                         // don't try to parse these for now
                         parser.skipChildren();
+                    } else if ("in_sync_allocations".equals(currentFieldName)) {
```

Comment:
This is just copy paste from the regular IndexMetadata parsing code. `in_sync_allocations` are required to exist for a restore to go through.